controller/lib/hal/hal.h

```diff
@@ -61,11 +62,16 @@ limitations under the License.
 
 #else // !TEST_MODE
 
-#ifndef AVR
-#error "When running without TEST_MODE, expecting AVR to be defined"
+#if !defined(ARDUINO_AVR_UNO) && !defined(ARDUINO_NUCLEO_L452RE)
+#error                                                                         \
+    "When running without TEST_MODE, expecting ARDUINO_AVR_UNO or ARDUINO_NUCLEO_L452RE to be defined"
 #endif
 #include <Arduino.h>
 
+#ifdef ARDUINO_AVR_UNO
```

Comment:
Is this only on Uno or on Nucleo too? Given we're now handling multiple defines, if possible, I think it'd be better to just ifdef on TEST_MODE everywhere below.